shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/controler/DatasetProcessingApiController.java

```diff
@@ -147,12 +164,72 @@ public ResponseEntity<Void> updateDatasetProcessing(
 			return new ResponseEntity<>(HttpStatus.NOT_FOUND);
 		}
 	}
-	
+
 	private void validate(BindingResult result) throws RestServiceException {
 		final FieldErrorMap errors = new FieldErrorMap(result);
 		if (!errors.isEmpty()) {
 			ErrorModel error = new ErrorModel(HttpStatus.UNPROCESSABLE_ENTITY.value(), "Bad arguments", new ErrorDetails(errors));
 			throw new RestServiceException(error);
 		}
 	}
+
+	@Override
+	public void massiveDownloadByProcessingId(
+			@Parameter(description = "ids of processing", required=true) @Valid
+			@RequestParam(value = "processingIds") List<Long> processingIds,
+			@Parameter(description = "outputs to extract") @Valid
+			@RequestParam(value = "resultOnly") boolean resultOnly,
+			HttpServletResponse response) throws RestServiceException {
+
+		List<DatasetProcessing> processingList = new ArrayList<>();
+		for (Long processingId : processingIds) {
+			DatasetProcessing processing = null;
+			try {
+				if(processingId == null){
+					throw new Exception();
+				}
+				processing = datasetProcessingService.findById(processingId).get();
+				processingList.add(processing);
+			}catch (Exception e) {
+				throw new RestServiceException(
+						new ErrorModel(HttpStatus.FORBIDDEN.value(), processingId + " is not a valid processing id."));
+			}
+
+			if(!Objects.equals(processing.getComment(), "comete_moelle/0.1")){
```

Comment:
I would add a filter param here, we can not keep this in the general code, so the caller decides, what to filter out